nvtabular/workflow/node.py

```diff
@@ -199,19 +269,39 @@ def __sub__(self, other):
         -------
         WorkflowNode
         """
-        if isinstance(other, WorkflowNode):
-            to_remove = set(other.output_columns)
-        elif isinstance(other, str):
-            to_remove = {other}
-        elif isinstance(other, collections.abc.Sequence):
-            to_remove = set(other)
+
+        if isinstance(self.op, internal.SubsetColumns):
+            child = self
         else:
-            raise ValueError(f"Expected WorkflowNode, str, or list of str. Got {other.__class__}")
-        new_columns = [c for c in self.output_columns if c not in to_remove]
-        child = WorkflowNode(new_columns)
-        child.parents = [self]
-        self.children.append(child)
-        child.op = internal.SubsetColumns(label=f"- {list(to_remove)}")
+            # Create a child node
+            child = WorkflowNode()
+            child.op = internal.SubsetColumns(label="-")
+
+            # Add self as a parent
+            self.children.append(child)
+            child.parents.append(self)
+
+        # The right operand becomes a dependency
```

Comment:
Longer term, we shouldn't be using dependencies for this, but `parents` are only allowed to be `WorkflowNodes`, not `ColumnSelectors`. Might be resolved by the proposed refactor to make selection into a node op?